### PR TITLE
Unify AI feature codebase and UX improvments

### PR DESCRIPTION
**Pull request by DonOmalVindula**
### Purpose
$subject

### Related Issues
- 

### Related PRs
- 

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: features/admin.branding.ai.v1/components/branding-ai-banner.tsx [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import SendOutlinedIcon from "@mui/icons-material/SendOutlined";
+import { Collapse } from "@mui/material";
```
[the remark this answers is not in the document]
Comment (DonOmalVindula) [reply features/admin.branding.ai.v1/components/branding-ai-banner.tsx:20]: These components are not available in the oxygen-ui

File: features/admin.branding.ai.v1/components/branding-ai-banner.tsx [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+                                        disabled={ !websiteUrl.trim() }
+                                    >
+                                        <SendOutlinedIcon
+                                            className={ `branding-ai-input-button-icon
+                                                ${ !websiteUrl.trim() && "disabled" }` }
+                                        />
+                                    </IconButton>
+                                ) : (
+                                    <Box>
+                                        <CircularProgress color="primary" size={ 25 } className="mr-2 mt-1" />
```
[the remark this answers is not in the document]
Comment (DonOmalVindula) [reply features/admin.branding.ai.v1/components/branding-ai-banner.tsx]: Resolved with af81c7fe8be01875fe8a2953898ed90a4e375ff9

File: features/admin.branding.ai.v1/components/branding-ai-banner.tsx [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+                            >
+                                <Trans i18nKey={ "extensions:common.learnMore" }>
+                                    Learn more
+                                </Trans>
+                            </DocumentationLink>
+                        </>
+                    ) }
+                    aiText={ t("branding:ai.title") }
+                    titleLabel={ (
+                        <Chip
```
[the remark this answers is not in the document]
Comment (DonOmalVindula) [reply features/admin.branding.ai.v1/components/branding-ai-banner.tsx:118]: Let's not add beta configs to deployment.config as they are not deployment specific. Rather product specific

File: features/admin.branding.ai.v1/components/branding-ai-banner.tsx [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+import useAIBrandingPreference from "../hooks/use-ai-branding-preference";
+import useGenerateAIBrandingPreference,
+{ GenerateAIBrandingPreferenceFunc } from "../hooks/use-generate-ai-branding-preference";
+import { BannerState } from "../models/types";
+import "./branding-ai-banner.scss";
+
+/**
+ * Branding AI banner component.
+ */
+export const BrandingAIBanner: FunctionComponent = (): ReactElement => {
```
Comment (brionmario) [inline features/admin.branding.ai.v1/components/branding-ai-banner.tsx:44]: Shall we use default exports?

Comment (DonOmalVindula) [reply features/admin.branding.ai.v1/components/branding-ai-banner.tsx]: Fixed with 2b12f3be4e49de7756b36bf0797366b0376b6b10

File: features/common.ai.v1/components/ai-banner-tall.tsx [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+    from "../../themes/wso2is/assets/images/illustrations/ai-banner-input-background-tall.png";
+import AIBot
+    from "../../themes/wso2is/assets/images/illustrations/ai-bot.svg";
+import "./ai-banner.scss";
+import "./ai-banner-tall.scss";
+
+/**
+ * Interface for the AI banner component.
+ */
+interface AIBannerTallProps extends IdentifiableComponentInterface {
```
Comment (brionmario) [inline features/common.ai.v1/components/ai-banner-tall.tsx:39]:
```suggestion
export interface AIBannerTallProps extends IdentifiableComponentInterface {
```

File: features/common.ai.v1/components/ai-banner-tall.tsx [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+    title?: ReactElement;
```
Comment (brionmario) [inline features/common.ai.v1/components/ai-banner-tall.tsx:40]: Add TSDoc.

File: features/common.ai.v1/components/ai-banner.tsx [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+import { IdentifiableComponentInterface } from "@wso2is/core/models";
+import React, { ReactElement } from "react";
+import AIText from "./ai-text";
+import AIBannerBackground from "../../themes/wso2is/assets/images/illustrations/ai-banner-background-white.svg";
+import "./ai-banner.scss";
+
+/**
+ * Interface for the AI banner component.
+ */
+interface AIBannerProps extends IdentifiableComponentInterface {
```
Comment (brionmario) [inline features/common.ai.v1/components/ai-banner.tsx:31]:
```suggestion
export interface AIBannerProps extends IdentifiableComponentInterface {
```

File: features/admin.branding.ai.v1/components/branding-ai-banner.tsx [hunk 8]
```diff
@@ -0,0 +1,180 @@
...
+import useAIBrandingPreference from "../hooks/use-ai-branding-preference";
+import useGenerateAIBrandingPreference,
+{ GenerateAIBrandingPreferenceFunc } from "../hooks/use-generate-ai-branding-preference";
+import { BannerState } from "../models/types";
+import "./branding-ai-banner.scss";
+
+/**
+ * Branding AI banner component.
+ */
+const BrandingAIBanner: FunctionComponent = (): ReactElement => {
```
Comment (brionmario) [inline features/admin.branding.ai.v1/components/branding-ai-banner.tsx:44]: Add `data-componentid`.

**Review by brionmario — APPROVED** [review]

**Pull request merged**